cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client_test.go

```diff
@@ -568,3 +574,207 @@ func TestUserAgent(t *testing.T) {
 	assert.NoError(t, err)
 	mock.AssertExpectationsForObjects(t, server)
 }
+
+// NOTE: pagination operations can't be tested with context timeouts as it's not possible
```

Comment:
FWIW, I think it still makes sense to have timeouts for paginated calls (for the whole operation). However, they were unbound before, so let's keep as is for now.